pydevconsole.py

```diff
@@ -356,12 +359,33 @@ def start_server(host, port, client_port):
     process_exec_queue(interpreter)
 
 
+def get_ipython_hidden_vars_dict():
+    useful_ipython_vars = ['_', '__']
+
+    try:
+        if IPYTHON and hasattr(__builtin__, 'interpreter'):
+            pydev_interpreter = get_interpreter().interpreter
+            if hasattr(pydev_interpreter, 'ipython') and hasattr(pydev_interpreter.ipython, 'user_ns_hidden'):
+                try:
+                    user_hidden_dict = pydev_interpreter.ipython.user_ns_hidden
+                except:
+                    user_hidden_dict = dict([(key, val) for key, val in pydev_interpreter.ipython.user_ns.items()
+                                    if key in pydev_interpreter.ipython.user_ns_hidden])
```

Comment:
Yes, it was done in order to support Python 1.x, where `user_ns_hidden` was a set with hidden variables, not a dictionary. But, of course, you're right it's implemented incorrectly.